lib/salus/scanners/gitleaks.rb

```diff
@@ -0,0 +1,109 @@
+require "salus/scanners/base"
+require "json"
+
+# Gitleaks scanner integration. Detects hardcoded secrets like passwords,
+# api keys, and tokens in git repos.
+# https://github.com/zricethezav/gitleaks
+
+module Salus::Scanners
+  class Gitleaks < Base
+    def run
+      Dir.chdir(@repository.path_to_repo) do
+        copts = config_options
+        shell_return = run_shell(
+          'gitleaks '\
+          '--path . '\
+          '--format sarif '\
+          '--report /dev/stdout '\
+          '--leaks-exit-code 0 ' +
+          copts
+        )
+
+        report_stderr(shell_return.stderr)
+
+        # A non-zero exit code is definitely an error
+        if !shell_return.success?
+          report_stdout(shell_return.stdout)
+          report_error("Call to gitleaks failed", {
+                         status: shell_return.status,
+                         stderr: shell_return.stderr.split("\n").last
+                       })
+          return report_failure
+        end
+
+        # Otherwise handle the SARIF output
+        # As of 2.7.1, no report is generated if no secrets were found
+        sarif = if shell_return.stdout.empty?
+                  {
+                    'runs' => []
+                  }
+                else
+                  JSON.parse(shell_return.stdout)
+                end
+
+        report_runs(sarif['runs'])
+      end
+    end
+
+    def should_run?
+      true # Always run on the provided folder
```

Comment:
What happens if you don't specify any configs for Gitleaks?
Have you tested this out with `docker build ... local_salus`, then `docker run ... local_salus`?